common/flatpak-context.c

```diff
@@ -1182,6 +1183,9 @@ option_filesystem_cb (const gchar *option_name,
   g_autofree char *fs = NULL;
   FlatpakFilesystemMode mode;
 
+  if (strstr(value, ":create") == NULL && strstr(value, ":rw") == NULL && strstr(value, ":ro") == NULL)
```

Comment:
This is duplicating the list of possible suffixes in `parse_filesystem_flags()`, which doesn't seem great: every time we add another one to the parsing, we have to duplicate it here. It also doesn't account for the possibility that the filesystem itself contains `:rw` (escaped as `\:rw`).

I'd prefer for `flatpak_context_parse_filesystem()` and `parse_filesystem_flags()` to have a boolean "out" parameter that means "a suffix was explicitly specified", something like this:

```
static char *
parse_filesystem_flags (const char            *filesystem,
                        gboolean               negated,
                        FlatpakFilesystemMode *mode_out,
                        gboolean              *explicit_suffix_out,
                        GError               **error)
{
...
  if (*p == ':')
    {
      if (explicit_suffix_out != NULL)
        *explicit_suffix_out = TRUE;

      suffix = p + 1;
...
    }
  else
    {
      if (explicit_suffix_out != NULL)
        *explicit_suffix_out = FALSE;
    }
}
```

and then use it something like this:

```
  gboolean explicit_suffix = FALSE;

  if (!flatpak_context_parse_filesystem (value, FALSE, &fs, &mode, &explicit_suffix, error))
    return FALSE;

  if (!explicit_suffix)
    g_warning ("It is recommended to explicitly use :rw for read-write access");
```

Reply:
I added the suffix check in `parse_filesystem_flags` the way you described it, I completely missed that it's possible to do it like that